spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/support/ResponseEntityDecoder.java

```diff
@@ -0,0 +1,64 @@
+
+package org.springframework.cloud.netflix.feign.support;
+
+import feign.FeignException;
+import feign.Response;
+import feign.codec.Decoder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.LinkedList;
+
+/**
+ * Decoder adds compatibility for Spring MVC's ResponseEntity to any
+ * other decoder via composition.
+ * @author chadjaros
+ */
+@Slf4j
+public class ResponseEntityDecoder implements Decoder {
```

Comment:
I'm using it within my code to allow return value type of ResponseEntity<ReturnType> on feign methods. This allows you to get the HTTP status and response headers. I figure I won't be the only one to want this.